frontend/src/packages/api/filesystem.ts

```diff
@@ -0,0 +1,92 @@
+import { CreateFilesystemEntryResponse, FilesystemEntry } from "./types/filesystem";
+import { APIError, EmptyAPIResponse } from "./types/general";
+
+// PostBody is an internal discriminated union of supported post bodies
+type PostBody = 
+    | { $type: "Record"; body: Record<string, string> }
+    | { $type: "FormData"; body: FormData };
+
+// TODO: not this
+// This was the best way I could figure out how to get actual integration tests to run in the frontend container
+let API_URL = "";
+
+export const configureApiUrl = (newUrl: string): void => { API_URL = newUrl };
+export const resetApiUrl = (): void => { API_URL = "" };
+
+// Only interface with the BE FS APIs via this class
+export class FilesystemAPI {
+    // GetEntityInfo retrieves all entity information for an FS entity given its ID
+    public static GetEntityInfo = (EntityID: string): Promise<FilesystemEntry | APIError> => 
+        FilesystemAPI.SendGetRequest<FilesystemEntry>(`/api/filesystem/info?EntityID=${EntityID}`);
+    
+    // GetRootInfo retrieves filesystem information for the file tree root
+    public static GetRootInfo = (): Promise<FilesystemEntry | APIError> =>
+        FilesystemAPI.SendGetRequest<FilesystemEntry>(`/api/filesystem/info`);
+
+    // CreateEntity constructs an editable un-published filesystem entry
+    public static CreateDocument = (Name: string, ParentID: string, ownerGroup = 1): Promise<CreateFilesystemEntryResponse | APIError> => 
```

Comment:
nitpicking but it should be `CreateDocument` not `CreateEntity` in the comment

Also, some filesystem endpoints are missing, is that intentional 👀